assets/js/workflow-editor/WorkflowEditor.tsx

```diff
@@ -0,0 +1,107 @@
+
+import type { WithActionProps } from '#/react/lib/with-props';
+import React from 'react';
+import WorkflowDiagram from '../workflow-diagram/WorkflowDiagram';
+import { useWorkflowStore, type ChangeArgs, type WorkflowProps } from './store';
+import { randomUUID } from '../common';
+import { DEFAULT_TEXT } from '../editor/Editor';
+
+const createNewWorkflow = (): Required<ChangeArgs> => {
+  const triggers = [
+    {
+      id: randomUUID(),
+      type: 'webhook' as 'webhook',
+    },
+  ];
+  const jobs = [
+    {
+      id: randomUUID(),
+      name: 'New job',
+      adaptor: '@openfn/language-common@latest',
+      body: DEFAULT_TEXT,
+    },
+  ];
+
+  const edges = [
+    {
+      id: randomUUID(),
+      source_trigger_id: triggers[0].id,
+      target_job_id: jobs[0].id,
+      condition_type: 'always',
+    },
+  ];
+  return { triggers, jobs, edges };
+};
+
+export const WorkflowEditor: WithActionProps<{ selection: string }> = (props) => {
+  const { add, setState, getItem } = useWorkflowStore();
+  const [workflowLoadParamsStart, setParamsStart] = React.useState<number>(null);
+  const [selection, setSelection] = React.useState(props.selection);
+  const onSelectionChange = (id?: string) => {
+    console.debug('onSelectionChange', id);
+
+    const currentUrl = new URL(window.location.href);
+    const nextUrl = new URL(currentUrl);
+
+    const idExists = getItem(id);
+    if (!idExists) {
+      nextUrl.searchParams.delete('s');
+      nextUrl.searchParams.delete('m');
+      nextUrl.searchParams.set('placeholder', 'true');
+    } else {
+      nextUrl.searchParams.delete('placeholder');
+      if (!id) {
+        console.debug('Unselecting');
+
+        nextUrl.searchParams.delete('s');
+        nextUrl.searchParams.delete('m');
+      } else {
+        console.debug('Selecting', id);
+
+        nextUrl.searchParams.set('s', id);
+      }
+    }
+
+    if (
+      currentUrl.searchParams.toString() !== nextUrl.searchParams.toString()
+    ) {
+      props.navigate(nextUrl.toString());
+    }
+  }
+
+  React.useEffect(() => {
+    props.handleEvent('navigate', (e: any) => {
```

Comment:
Can we create a `useWorkflowSelectionHook` to track this? The URL manipulation here is surprising and unclean. Be nice to tidy it away into a hook (and maybe later that hook can use the store rather than the URL)

Reply:
moved into store. I doubt we want to wrap the event handlers in a hook.